stl/inc/chrono

```diff
@@ -5171,6 +5176,183 @@ namespace chrono {
+    concept _Chrono_parse_spec_callbacks = _Parse_align_callbacks<_Ty, _CharT>
+                                           && _Parse_width_callbacks<_Ty, _CharT>
+                                           && _Parse_precision_callbacks<_Ty, _CharT>
+                                           && _Width_adapter_callbacks<_Ty, _CharT>
+                                           && _Precision_adapter_callbacks<_Ty, _CharT>
+                                           && requires(_Ty _At, basic_string_view<_CharT> _Sv, _Align _Aln) {
+        { _At._On_conversion_spec(_CharT{}, _CharT{}) } -> same_as<void>;
+        { _At._On_lit_char(_CharT{}) } -> same_as<void>;
+    };
+    // clang-format on
+
+    template <class _CharT>
+    struct _Chrono_specs {
+        _CharT _Lit_char = _CharT{0}; // any character other than '{', '}', or '%'
+        char _Modifier   = '\0'; // either 'E' or 'O'
+        char _Type       = '\0';
+    };
+
+    template <class _CharT>
+    struct _Chrono_format_specs {
+        int _Width                   = 0;
+        int _Precision               = -1;
+        int _Dynamic_width_index     = -1;
+        int _Dynamic_precision_index = -1;
+        _Align _Alignment            = _Align::_None;
+        uint8_t _Fill_length         = 1;
+        // At most one codepoint (so one char32_t or four utf-8 char8_t)
+        _CharT _Fill[4 / sizeof(_CharT)] = {_CharT{' '}};
+        // recursive definition in grammar, so could have any number of these with literal chars
+        vector<_Chrono_specs<_CharT>> _Chrono_specs_list;
+    };
+
+    // Model of _Chrono_parse_spec_callbacks that fills a _Chrono_format_specs with the parsed data
+    template <class _CharT>
+    class _Chrono_specs_setter {
+    public:
+        constexpr explicit _Chrono_specs_setter(_Chrono_format_specs<_CharT>& _Specs_) : _Specs(_Specs_) {}
+
+        // same as _Specs_setter
+        constexpr void _On_align(_Align _Aln) {
+            _Specs._Alignment = _Aln;
+        }
+
+        // same as _Specs_setter
+        constexpr void _On_fill(basic_string_view<_CharT> _Sv) {
+            if (_Sv.size() > _STD size(_Specs._Fill)) {
+                _THROW(format_error("Invalid fill (too long)."));
+            }
+
+            const auto _Pos = _STD _Copy_unchecked(_Sv._Unchecked_begin(), _Sv._Unchecked_end(), _Specs._Fill);
+            _STD fill(_Pos, _STD end(_Specs._Fill), _CharT{});
+            _Specs._Fill_length = static_cast<uint8_t>(_Sv.size());
+        }
+
+        constexpr void _On_width(int _Width) {
+            _Specs._Width = _Width;
+        }
+
+        constexpr void _On_precision(int _Prec) {
+            _Specs._Precision = _Prec;
+        }
+
+        constexpr void _On_conversion_spec(_CharT _Modifier, _CharT _Type) {
+            // NOTE: same performance note from _Basic_format_specs also applies here
+            const char _Char_mod  = static_cast<char>(_Modifier);
+            const char _Char_type = static_cast<char>(_Type);

```

Comment:
Note (no action required): this probably breaks EBCDIC, we don't really care about EBCDIC though.

Reply:
We don't support any EBCDIC platforms.